src/app/clusters/pump-configuration-and-control-server/pump-configuration-and-control-server.cpp

```diff
@@ -18,19 +18,321 @@
+        Attributes::EffectiveControlMode::Set(endpoint, controlMode);
+        pumpStatus.Clear(PumpStatus::kRemoteFlow);
+        pumpStatus.Clear(PumpStatus::kRemotePressure);
+        pumpStatus.Clear(PumpStatus::kRemoteTemperature);
+    }
+    break;
+    }
+    Attributes::PumpStatus::Set(endpoint, pumpStatus);
+}
+
+CHIP_ERROR PumpConfigurationAndControlAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    emberAfDebugPrintln("Reading from PCC");
+
+    VerifyOrDie(aPath.mClusterId == PumpConfigurationAndControl::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PumpConfigurationAndControlAttrAccess::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+{
+    emberAfDebugPrintln("Writing to PCC");
+
+    VerifyOrDie(aPath.mClusterId == PumpConfigurationAndControl::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+} // namespace
+
+// SDK Callbacks
+
+template <typename T1, typename T2>
+bool IsFeatureSupported(EndpointId endpoint, EmberAfStatus (*getFn1)(chip::EndpointId endpointId, T1 & value),
+                        EmberAfStatus (*getFn2)(chip::EndpointId endpointId, T2 & value))
+{
+    EmberAfStatus status;
+
+    T1 value1;
+    T2 value2;
+
+    status = getFn1(endpoint, value1);
+    if (status == EMBER_ZCL_STATUS_SUCCESS)
+    {
+        if (!value1.IsNull())
+        {
+            status = getFn2(endpoint, value2);
+            if (status == EMBER_ZCL_STATUS_SUCCESS)
+            {
+                if (!value2.IsNull())
+                {
+                    return true;
+                }
+            }
+        }
+    }
+
+    return false;
+}
 
 void emberAfPumpConfigurationAndControlClusterServerInitCallback(EndpointId endpoint)
 {
+    bool constPressureSupported     = false;
+    bool constPropPressureSupported = false;
+    bool constFlowSupported         = false;
+    bool constTemperatureSupported  = false;
+    bool constSpeedSupported        = false;
+
     emberAfDebugPrintln("Initialize PCC Server Cluster [EP:%d]", endpoint);
-    // TODO
+
+    // Determine the internal feature set of the pump, depending on the pump
+    // specific attributes available, and their values. If an attribute is
+    // present/available, then there is a possibility for the associated
+    // feature being present as well. But we will have to distinguis between
+    // the attributes being available and null also. At this point (init)
+    // we can only examine the list of attributes available, and then detect
+    // if they each have a NonNull value. Later on, when the pump application
+    // has finished its init process, it might setup these attributevalues
+    // to something NonNull, and then we must re-calcualte the feature set.
```

Comment:
Let us clearly state that this is due to a missing feature map, which should be added in future versions of PCC.

Reply:
I agree, and I have done so :-)